api/handlers/listhandler.py

```diff
@@ -474,6 +484,22 @@ def post(self, cont_name, list_name, **kwargs):
 
         return upload.process_upload(self.request, upload.Strategy.targeted, container_type=cont_name, id_=_id, origin=self.origin)
 
+    def delete(self, cont_name, list_name, **kwargs):
+        # Overriding base class delete to audit action before completion
+        _id = kwargs.pop('cid')
+        permchecker, storage, _, _, keycheck = self._initialize_request(cont_name, list_name, _id, query_params=kwargs)
+
+        permchecker(noop)('DELETE', _id=_id, query_params=kwargs)
+        self.log_user_access(AccessType.delete_file, cont_name=cont_name, cont_id=_id)
+        try:
+            result = keycheck(storage.exec_op)('DELETE', _id, query_params=kwargs)
+        except APIStorageException as e:
+            self.abort(400, e.message)
+        if result.modified_count == 1:
+            return {'modified': result.modified_count}
+        else:
+            self.abort(404, 'Element not removed from list {} in container {} {}'.format(storage.list_name, storage.cont_name, _id))
+
```

Comment:
It's just a logging version of the base delete method here: https://github.com/scitran/core/pull/644/files/c7c042b41b7c19977e02f3da32016107604066fd#diff-2b6de8dffa100181ea2c9b85e6e5f50cR166

`listhandler.py` has been relatively untouched for over a year, it might be time to see how we can optimize it after our growth over the last year.